solver/src/http_solver/settlement.rs

```diff
@@ -0,0 +1,204 @@
+            } else if model.balance_update_1 == 0 && model.balance_update_2 == 0 {
+                continue;
+            } else {
+                return Err(anyhow!("invalid uniswap update {:?}", model));
+            };
+        let interactions = amm.settlement_handling.settle(input, output);
+        settlement.interactions.extend(interactions);
+    }
+    Ok(())
+}
+
+fn set_prices(
+    model: &SettledBatchAuctionModel,
+    tokens: &HashMap<String, H160>,
+    settlement: &mut Settlement,
+) -> Result<()> {
+    for (index, &Price(price)) in model.prices.iter() {
+        let token = tokens
+            .get(index.as_str())
+            .ok_or_else(|| anyhow!("invalid token {}", index))?;
+        let token_used_in_trade = settlement
+            .trades
+            .iter()
+            .any(|trade| *token == trade.order.sell_token || *token == trade.order.buy_token);
+        dbg!(token, token_used_in_trade);
+        if token_used_in_trade {
+            ensure!(price.is_finite() && price > 0.0, "invalid price {}", price);
+            let price = U256::from_f64_lossy(price);
+            settlement.clearing_prices.insert(*token, price);
+        }
+    }
+    Ok(())
+}
+
+fn i128_abs_to_u256(i: i128) -> U256 {
+    // TODO: use `unsigned_abs` once it is stable in next compiler version
+    // until then we need this check because the most negative value can not be `abs`ed because it
+    // it the most positive value plus 1.
+    if i == i128::MIN {
+        (i128::MAX as u128 + 1).into()
+    } else {
+        (i.abs() as u128).into()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{
+        http_solver::model::{ExecutedOrderModel, UpdatedUniswapModel},
+        liquidity::{MockAmmSettlementHandling, MockLimitOrderSettlementHandling},
+        settlement::{Interaction, Trade},
+    };
+    use maplit::hashmap;
+    use mockall::predicate::eq;
+    use model::{order::OrderCreation, TokenPair};
+    use std::sync::Arc;
+
+    #[derive(Debug)]
+    struct NoopInteraction;
+    impl Interaction for NoopInteraction {
+        fn encode(&self, _writer: &mut dyn std::io::Write) -> Result<()> {
+            Ok(())
+        }
+    }
+
+    #[test]
+    fn convert_settlement_() {
+        let t0 = H160::from_low_u64_be(0);
+        let t1 = H160::from_low_u64_be(1);
+        let tokens = hashmap! { "t0".to_string() => t0, "t1".to_string() => t1 };
+
+        let mut limit_handling = MockLimitOrderSettlementHandling::new();
+        limit_handling.expect_settle().returning(move |_| {
+            (
+                Some(Trade {
+                    order: OrderCreation {
+                        sell_token: t0,
+                        buy_token: t1,
+                        sell_amount: 1.into(),
+                        buy_amount: 2.into(),
+                        ..Default::default()
+                    },
+                    ..Default::default()
+                }),
+                vec![],
```

Comment:
nit; could return a noop interaction to test if interactions here are also part of the settlement